### ISSUE TITLE
1.17.1 - Entity Death Logs

### ISSUE THREAD

**Issue by swim709** [PaperMC/Paper#6106]
### Expected behavior

1.17.1 introduced this feature:
-Deaths of named mobs are now logged

### Observed/Actual behavior

Unnamed mobs killed with a named sword are getting logged to the console

### Steps/models to reproduce

Run Paper build 79. Name a weapon with a name tag. Kill a mob.

### Plugin list

ArmoredElytra, BannerMaker, BetterChairs, BetterFarming, Brewery, ChatBubbles, ChestSort, Chunkmaster, CoordinatesHud, CoreProtect, DiscordSRV, DropHeads, dynmap, EasyBackup, EasyBackupCore, Essentials, EssentialsChat, ExoticGarden, GriefPrevention, HolographicDisplays, HotbarPets, InventoryRollback, LuckPerms, Magic, mcMMO, NetherCoords*, OpenInv, Plan, Playtimes*, sleep-most, Slimefun, UltimateShops, Vault, ViaVersion, WorldBorder, WorldEdit, WorldGuard, XpBottles

### Paper version

git-Paper-79

### Agreements

- [X] I am running the latest version of Paper available from https://papermc.io/downloads.
- [X] I have searched for and ensured there isn't already an open issue regarding this.
- [X] My version of Minecraft is supported by Paper.

**Comment by CaileanTV** [comment]
i do get constant "Named entity EntityEnderman['Enderman'/7380, uuid='c9288922-0aec-46d0-b91b-774a927139b2', l='ServerLevel[S_the_end]', x=577.70, y=69.06, z=-900.04, cpos=[36, -57], tl=677, v=true, rR=null]: died: Enderman was slain by [playername]" from our enderman farm... so 100% unnamed entities...

**Comment by Chew** [comment]
Build 79 was 1.17... marking as a bug though

**Comment by Machine-Maker** [comment]
> Name a weapon with a name tag. Kill a mob.

@swim709 I assume you meant name a mob with a name tag right?

**Comment by swim709** [comment]
> 
> 
> > Name a weapon with a name tag. Kill a mob.
> 
> @swim709 I assume you meant name a mob with a name tag right?

No I actually did mean a weapon. At the time it seemed that was causing it because none of the mobs were named. However I realized after that it actually had nothing to do with the weapon being named, it was just happening for any mobs being killed. I just updated to build 88 and the problem seems to be gone. I must have mistakenly wrote build 79 in the original report, I was using build 83 at the time.
